fvm/env.go

```diff
@@ -535,12 +535,16 @@ func (e *hostEnv) Logs() []string {
 	return e.logs
 }
 
-func (e *hostEnv) Hash(data []byte, hashAlgorithm runtime.HashAlgorithm) ([]byte, error) {
+func (e *hostEnv) Hash(data []byte, tag string, hashAlgorithm runtime.HashAlgorithm) ([]byte, error) {
```

Comment:
I think Cadence supports 2 hashing APIs, a hash with tag, and a hash without tag. 
Is there another FVM function for the hash without tag?

Reply:
@turbolent I have not found the hashing API without a tag. The only usage of the `interface.Hash` function I found is here: https://github.com/onflow/cadence/blob/ca1e6400714f4b412688add73ddaf72b945c9562/runtime/stdlib/crypto.go#L221 which includes the tag. Is there a usage without the tag?

This is also the the only `hash` function on the interface.

Reply:
There are two hash functions available to users:

https://github.com/onflow/cadence/blob/ca1e6400714f4b412688add73ddaf72b945c9562/runtime/stdlib/contracts/crypto.cdc#L25-L31

Note how the hash function without the tag delegates to the one with the tag, passing an empty string.

So in both cases the `hash` function in the Go code is called:

https://github.com/onflow/cadence/blob/ca1e6400714f4b412688add73ddaf72b945c9562/runtime/stdlib/contracts/crypto.cdc#L16-L20

Further, Cadence calls the host environments `Hash` function through the runtime interface:

https://github.com/onflow/cadence/blob/bcc383fdcc00efec5f27782191382f6d22ca1253/runtime/interface.go#L104-L105

Reply:
Thanks @turbolent! That clarifies why there is only one hashing function on the FVM side. 
@janezpodhostnik your changes are valid in this case, as we're still allowing empty tags.